lib/safari-launcher.js

```diff
@@ -0,0 +1,105 @@
+import xcode from 'appium-xcode';
+import { fs } from 'appium-support';
+import { exec } from 'teen_process';
+import logger from './logger';
+import path from 'path';
+
+
+const SAFARI_LAUNCHER_DIR = path.resolve(__dirname, '..', '..',
+                                              'build', 'SafariLauncher');
+const SAFARI_LAUNCHER_APP_FILE = path.resolve(__dirname, '..', '..',
+                                              'build', 'SafariLauncher',
+                                              'SafariLauncher.app');
+const SAFARI_LAUNCHER_REPO = path.resolve(__dirname, '..', '..',
+                                          'node_modules', 'safari-launcher');
+const SAFARI_LAUNCHER_CONFIG_FILE = path.resolve(SAFARI_LAUNCHER_REPO, 'build.xconfig');
+const SAFARI_LAUNCHER_BUNDLE = 'com.bytearc.SafariLauncher';
+
+const sdks = ['iphoneos'];
+
+async function cleanApp (appRoot, sdk) {
+  logger.debug(`Cleaning SafariLauncher for ${sdk}`);
+  try {
+    await exec('xcodebuild', ['-sdk', sdk, 'clean'], {cwd: appRoot});
+  } catch (err) {
+    logger.error(err);
```

Comment:
this looks like a good place for `logger.errorAndThrow` or whatever it is called